cdap-ui/app/cdap/components/EntityTopPanel/index.js

```diff
@@ -26,31 +26,60 @@ export default class EntityTopPanel extends PureComponent {
     breadCrumbAnchorLink: PropTypes.string,
     breadCrumbAnchorLabel: PropTypes.string,
     title: PropTypes.string,
-    closeBtnAnchorLink: PropTypes.oneOfType([PropTypes.string, PropTypes.func])
+    entityIcon: PropTypes.string,
+    entityType: PropTypes.string,
+    closeBtnAnchorLink: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
+    historyBack: PropTypes.bool
   };
 
   renderBreadCrumnAnchorLink = () => {
-    if (!this.props.breadCrumbAnchorLink) {
+    if (!this.props.breadCrumbAnchorLink && !this.props.historyBack) {
       return null;
     }
+
+    let link = this.props.breadCrumbAnchorLink;
+    let Tag = Link;
+
+    if (this.props.historyBack) {
+      link = document.referrer;
```

Comment:
This will refresh the whole page right? Even if I am coming from entity detailed view?

Reply:
@ajainarayanan yes it will. Do you have a better suggestion?

Reply:
`history.back()` will be state transition but it comes with its own problems.. my only concern is this will be an entire page refresh.